compiler/ir.py

```diff
@@ -849,11 +850,11 @@ def apply_round_robin_parallelization_to_node(
         #  currently, this cannot be done since splitter etc. would be added...
         aggregator_spec = parallelizer.get_aggregator_spec()
         if aggregator_spec.is_aggregator_spec_adj_lines_merge():
-            raise Exception("adj_lines_merge not yet implemented in PaSh")
```

Comment:
Konstantinos (from old branch): These 3 errors are actually general exceptions. Or rather NotImplemented errors, so I would make a new error type and catch them separately.